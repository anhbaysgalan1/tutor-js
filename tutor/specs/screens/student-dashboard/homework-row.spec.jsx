import React from 'react';
import SnapShot from 'react-test-renderer';
import moment from 'moment-timezone';
import chronokinesis from 'chronokinesis';
import Factory from '../../factories';
import HomeworkRow from '../../../src/screens/student-dashboard/homework-row';


describe('Homework Row', function() {
  let props;

  beforeEach(function() {
    chronokinesis.travel(new Date('2017-10-14T12:00:00.000Z'));
    moment.tz.setDefault('America/Chicago');
    props = {
      event: Factory.studentDashboardTask({ type: 'homework' }),
      course: Factory.course(),
    };
  });

  it('renders with completed count', function() {
    props.event.correct_exercise_count = null;
    expect(SnapShot.create(<HomeworkRow {...props} />).toJSON()).toMatchSnapshot();
  });

  it('renders with correct count', function() {
    props.event.correct_exercise_count = 0;
    expect(SnapShot.create(<HomeworkRow {...props} />).toJSON()).toMatchSnapshot();
    props.event.correct_exercise_count = 2;
    expect(SnapShot.create(<HomeworkRow {...props} />).toJSON()).toMatchSnapshot();
  });

});
