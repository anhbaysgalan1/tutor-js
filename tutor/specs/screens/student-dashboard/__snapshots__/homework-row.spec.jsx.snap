// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homework Row renders with completed count 1`] = `
<a
  aria-label="Work on homework: turn-key leverage content"
  className="task row homework workable"
  data-event-id={1}
  href="/course/1/task/1"
  onClick={[Function]}
  onKeyDown={undefined}
  tabIndex={0}
>
  <div
    className="column-icon col-sm-1 col-xs-2"
  >
    <i
      aria-label="homework icon"
      className="icon icon-lg icon-homework"
    />
  </div>
  <div
    className="title col-sm-6 col-xs-10"
  >
    turn-key leverage content
  </div>
  <div
    className="feedback col-sm-3 col-xs-5"
  >
    <span>
      <span
        className="tour-anchor"
        data-tour-anchor-id="assignment-progress-status"
        id="assignment-progress-status"
      >
        2/3 answered
      </span>
    </span>
  </div>
  <div
    className="due-at col-sm-2 col-xs-5"
  >
    <time>
      Oct 11, 7:00am
    </time>
  </div>
</a>
`;

exports[`Homework Row renders with correct count 1`] = `
<a
  aria-label="Work on homework: out-of-the-box embrace supply-chains"
  className="task row homework workable"
  data-event-id={2}
  href="/course/2/task/2"
  onClick={[Function]}
  onKeyDown={undefined}
  tabIndex={0}
>
  <div
    className="column-icon col-sm-1 col-xs-2"
  >
    <i
      aria-label="homework icon"
      className="icon icon-lg icon-homework"
    />
  </div>
  <div
    className="title col-sm-6 col-xs-10"
  >
    out-of-the-box embrace supply-chains
  </div>
  <div
    className="feedback col-sm-3 col-xs-5"
  >
    <span>
      <span
        className="tour-anchor"
        data-tour-anchor-id="assignment-progress-status"
        id="assignment-progress-status"
      >
        0/10 correct
      </span>
    </span>
    <span
      className="tour-anchor"
      data-tour-anchor-id="about-late-icon"
      id="about-late-icon"
    >
      <i
        className="info late"
        onBlur={[Function]}
        onClick={null}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      />
    </span>
  </div>
  <div
    className="due-at col-sm-2 col-xs-5"
  >
    <time>
      Oct 11, 7:00am
    </time>
  </div>
</a>
`;

exports[`Homework Row renders with correct count 2`] = `
<a
  aria-label="Work on homework: out-of-the-box embrace supply-chains"
  className="task row homework workable"
  data-event-id={2}
  href="/course/2/task/2"
  onClick={[Function]}
  onKeyDown={undefined}
  tabIndex={0}
>
  <div
    className="column-icon col-sm-1 col-xs-2"
  >
    <i
      aria-label="homework icon"
      className="icon icon-lg icon-homework"
    />
  </div>
  <div
    className="title col-sm-6 col-xs-10"
  >
    out-of-the-box embrace supply-chains
  </div>
  <div
    className="feedback col-sm-3 col-xs-5"
  >
    <span>
      <span
        className="tour-anchor"
        data-tour-anchor-id="assignment-progress-status"
        id="assignment-progress-status"
      >
        2/10 correct
      </span>
    </span>
    <span
      className="tour-anchor"
      data-tour-anchor-id="about-late-icon"
      id="about-late-icon"
    >
      <i
        className="info late"
        onBlur={[Function]}
        onClick={null}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      />
    </span>
  </div>
  <div
    className="due-at col-sm-2 col-xs-5"
  >
    <time>
      Oct 11, 7:00am
    </time>
  </div>
</a>
`;
